source/matplot/util/common.h

```diff
@@ -46,6 +46,8 @@ namespace matplot {
 
     template <class T> std::string num2str(Arithmetic<T> num) {
         std::ostringstream ss;
+        ss.precision(5);
```

Comment:
I'm not sure I completely understand the motivation for `precision(5)` at other classes, but it doesn't look like a sensible default for this function. 

For instance, 

```python
str(2.5433653543232154254214321541)
```

gives me 

```python
'2.5433653543232153'
```

in python.